File: src/main/java/de/weimarnetz/registrator/repository/RegistratorRepository.java
package de.weimarnetz.registrator.repository;

import java.util.List;

import org.springframework.data.repository.CrudRepository;

import de.weimarnetz.registrator.model.Node;

public interface RegistratorRepository extends CrudRepository<Node, Long> {

    Node findByNumberAndNetwork(int number, String network);

    Node findByNetworkAndMac(String network, String mac);

    List<Node> findAllByNetwork(String network);

    List<Node> findAllByNetworkOrderByLastSeenDesc(String network);
}

File: src/main/java/de/weimarnetz/registrator/repository/RegistratorRepository.kt
package de.weimarnetz.registrator.repository

import de.weimarnetz.registrator.model.Node
import org.springframework.data.repository.CrudRepository

interface RegistratorRepository : CrudRepository<Node, Long> {
    fun findByNumberAndNetwork(number: Int, network: String): Node?
    fun findByNetworkAndMac(network: String, mac: String): Node?
    fun findAllByNetwork(network: String): List<Node>
    fun findAllByNetworkOrderByLastSeenDesc(network: String): List<Node>
}